src/commands/mod.rs

```diff
@@ -680,4 +909,93 @@ settings:
 
         assert_eq!(contents, expected_contents);
     }
+
+    #[test]
+    fn test_validate_file_when_file_non_existent_should_return_error() {
+        let dir = tempdir().unwrap();
+        let file_path = dir.path().join("test.yaml");
+
+        let result = EdgeAppManifest::validate_file(&file_path);
+        assert!(result.is_err(), "Expected an error for non-existent file");
+    }
+
+    #[test]
+    fn test_validate_file_when_file_valid_should_return_true() {
+        let dir = tempdir().unwrap();
+        let file_name = "test.yaml";
+        let content = r#"---
+app_id: test_app
+settings:
+  username:
+    type: string
+    default_value: stranger
+    title: username
+    optional: true
+    help_text: An example of a setting that is used in index.html
+"#;
+
+        write_to_tempfile(&dir, file_name, content);
+        let file_path = dir.path().join(file_name);        
+        assert_eq!(EdgeAppManifest::validate_file(&file_path).unwrap(), true);
+    }
+
+    #[test]
+    fn test_validate_file_when_missing_field_should_return_false() {
+        let dir = tempdir().unwrap();
+        let file_name = "test.yaml";
+        let content = r#"---
+app_id: test_app
+settings:
+  username:
+    type: string
+    default_value: stranger
+    title: username
+    help_text: An example of a setting that is used in index.html
+"#;
+
+        write_to_tempfile(&dir, file_name, content);
+        let file_path = dir.path().join(file_name);        
+        assert_eq!(EdgeAppManifest::validate_file(&file_path).unwrap(), false);
+    }
+
+    #[test]
+    fn test_validate_file_when_empty_field_should_return_false() {
+        let dir = tempdir().unwrap();
+        let file_name = "test.yaml";
```

Comment:
it's screenly.yml

Reply:
It doesn't matter for tests but I can change it